esmvaltool/diag_scripts/climate_metrics/create_table.py

```diff
@@ -86,7 +89,10 @@ def _calculate_statistic(data_frame, stat_func, exclude_datasets):
         )
         series_to_append.append(series)
     for series in series_to_append:
-        data_frame = data_frame.append(series)
+        data_frame = pd.concat([
+            data_frame,
+            series.to_frame().T,
+        ])
```

Comment:
This is an unfortunate pattern for concatenation because it requires repeated allocations and copying. Would something like the following work?
```suggestion
    data_frame = pd.concat([series.to_frame().T for series in series_to_append])
```
Possibly even without the `[]`, turning it into a generator leaving any additional allocation up to `pd.concat`.

Reply:
Very good point Klaus, I changed this (and also for other appearances of `pd.concat`) in https://github.com/ESMValGroup/ESMValTool/pull/3209/commits/66c63a1e106535b62bd5555855390325d86a0ed4